renderdoc/driver/vulkan/wrappers/vk_queue_funcs.cpp

```diff
@@ -1108,7 +1119,7 @@ VkResult WrappedVulkan::vkQueueSubmit(VkQueue queue, uint32_t submitCount,
               state.mapFlushed = false;
             }
 
-            GetResourceManager()->MarkDirtyWithWriteReference(record->GetResourceID());
+            GetResourceManager()->MarkDirtyResource(record->GetResourceID());
```

Comment:
It's likely harmless either way because dirty resources are fixed at capture start, so by the time we reach here this is only relevant for later frames and in vulkan device memory is almost always considered dirty.